pkg/policy/repository.go

```diff
@@ -765,3 +764,26 @@ func (p *Repository) computePolicyEnforcementAndRules(securityIdentity *identity
 		return false, false, nil
 	}
 }
+
+func (p *Repository) UpdatePolicyMetrics(idnty *identity.Identity, isAdd bool) {
+
+	_, _, matchingRules := p.computePolicyEnforcementAndRules(idnty)
+
+	for _, v := range matchingRules {
+		tcount := 0
+		for _, value := range v.metadata.IdentitySelected {
+			if value == true {
+				tcount++
+			}
+		}
+		if tcount == 1 && idnty.ID != identity.ReservedIdentityHealth {
+			if isAdd == true && idnty.ReferenceCount == 1 {
```

Comment:
Identity reference count can also be > 1, e.g., when multiple endpoints share the same identity.